### PR TITLE
implement add and delete method

### PR DESCRIPTION
**Pull request by fabpot**
This PR was submitted on the symfony/ldap read-only repository by @L0rD59 and automatically moved to the main repository (closes symfony/ldap#1).

Hi

Can i have some review please ?

Thx.

### CODE REVIEW

File: src/Symfony/Component/Ldap/LdapClient.php [hunk 1]
```diff
@@ -54,6 +54,8 @@ public function __construct($host = null, $port = 389, $version = 3, $useSsl = f
         $this->useSsl = (bool) $useSsl;
         $this->useStartTls = (bool) $useStartTls;
         $this->optReferrals = (bool) $optReferrals;
+
+        $this->connect();
```
Comment (xabbuh) [inline src/Symfony/Component/Ldap/LdapClient.php]: I don't think it's a good idea to open the connection in the constructor.

Comment (L0rD59) [inline src/Symfony/Component/Ldap/LdapClient.php]: I don't know, the returned resource is required by 99% of ldap_* function.

If you want, i can put as was.

Comment (ChadSikorra) [inline src/Symfony/Component/Ldap/LdapClient.php]: I almost think there should be a config class or an array of options passed on the constructor. That way you could have something like an auto-connect option that could default to true, but could still be changed if someone wanted to.

Comment (csarrazi) [inline src/Symfony/Component/Ldap/LdapClient.php]: @ChadSikorra PR #17560 adresses this issue of the constructor.

@L0rD59, the connection should not be done in the constructor. Indeed, services which depend on the Ldap service may actually want to do checks before even trying to connect to the Ldap server. Like checking whether the user's password is empty or not before trying to bind against the server.

**Comment by csarrazi** [comment]
We should keep in mind that this PR won't be compatible with the next iteration of the component.

Also, this is a new feature, and thus should be implemented in the master branch instead.

**Comment by fabpot** [comment]
closing in favor of #17817

**Comment by L0rD59** [comment]
:+1:

**Pull request closed without merging**